tempodb/blocklist/poller_test.go

```diff
@@ -381,6 +381,132 @@ func TestTenantIndexPollError(t *testing.T) {
 	}, nil))
 }
 
+func TestBlockListBackendMetrics(t *testing.T) {
+	tests := []struct {
+		name                        string
+		list                        PerTenant
+		compactedList               PerTenantCompacted
+		testType                    string
+		expectedBackendObjectsTotal int
+		expectedBackendBytesTotal   uint64
```

Comment:
I think we should add similar fields for compacted objects meta just to make it explicit that we're testing those as well -- in the current scenario the total counts of objects/sizes are coincidentally lining up which can be confusing if the tests change in the future